wandb/sdk/internal/file_stream.py

```diff
@@ -373,18 +372,16 @@ def _thread_body(self):
 
             cur_time = time.time()
 
-            if ready_chunks and (
+            if (ready_chunks or uploaded) and (
                 finished or cur_time - posted_data_time > self.rate_limit_seconds()
             ):
                 posted_data_time = cur_time
-                posted_anything_time = cur_time
-                self._send(ready_chunks)
-                ready_chunks = []
+                if ready_chunks:
+                    self._send(ready_chunks)
+                    ready_chunks = []
 
-            if cur_time - posted_anything_time > self.heartbeat_seconds:
```

Comment:
Should we keep this `if` conditional so that we always heartbeat at a consistent rate, regardless of whether we've uploaded files or not?